### ISSUE TITLE
nodeenv try to download non existing tar.gz prebuilt under Cygwin

### ISSUE THREAD

**Issue by Lucas-C** [pre-commit/pre-commit#455]
Hi,

Strange issue: I suspect a recent change broke this as it used to work last week, on another Windows computer with Cygwin.

Bug reproduction: `pre-commit run` using e.g. https://github.com/Lucas-C/pre-commit-hooks-html v1.1.0

`pre-commit` execute the following command under the hood, a command that also fails if I execute it manually:
```
nodeenv --prebuilt /cygdrive/c/Users/admin/.pre-commit/repoYHJ85q/node_env-default
```
The error is the following:
```
urllib2.HTTPError: HTTP Error 404: Not Found
```
The `tar.gz` it tries to install is https://nodejs.org/dist/v7.2.1/node-v7.2.1-cygwin_nt-6.1-x64.tar.gz, which does not exist. My guess is that `nodeenv` should use the Windows prebuilts instead: https://nodejs.org/dist/v7.2.1/node-v7.2.1-win-x64.zip This is because `platform.system()` is used: https://github.com/ekalinin/nodeenv/blob/master/nodeenv.py#L503

I'm going to ask for help on the https://github.com/ekalinin/nodeenv project, but do you have any hint at what the root cause could be here ?

**Comment by asottile** [comment]
I'm actually surprised cygwin worked before, nothing seems changed in the last couple weeks in `nodeenv` and I don't see any changes to the files provided by nodejs either.  To be honest, I didn't know nodeenv even worked on windows/cygwin: #200 

I think you're probably right about the source of the issue.  Another solution might be to have the nodejs guys compile against cygwin though it's a difficult platform to target (since the packaging moves like arch).

**Comment by Lucas-C** [comment]
You're right, I must have mixed up things, it probably never worked.

I think there's no need to keep this ticket open, I'm tracking this in https://github.com/ekalinin/nodeenv/issues/178

**Comment by Lucas-C** [comment]
Could it be possible to use the latest nodeenv version to fix this please ? :)
https://github.com/ekalinin/nodeenv/releases/tag/1.2.0

**Comment by asottile** [comment]
Yeah! pre-commit doesn't limit to a specific version of nodeenv so you should be able to just `pip install --upgrade nodeenv`

**Comment by Lucas-C** [comment]
Great, thanks :)

**Comment by Lucas-C** [comment]
Bad news : NodeJS hooks still do not work :(

The issue comes from the `NPM_CONFIG_PREFIX` / `npm_config_prefix` environment variables:
https://github.com/pre-commit/pre-commit/blob/master/pre_commit/languages/node.py#L22-L23

Among the fixes I made in `nodeenv` to add support for Cygwin, I had to alter those 2 env variables:
https://github.com/ekalinin/nodeenv/blob/master/nodeenv.py#L850

Are you OK if I send a PR adding a `if cygwin` conditional in `node.py:get_env_patch`

**Comment by asottile** [comment]
seems fine to me :)

**Comment by Lucas-C** [comment]
There is still an issue with this.
I think I took a shortcut when I tested the previous PR (I must have skipped testing a full repo reinstall), and it definitively still does not work.
Sorry about that :(

This time it is because the `NPM_CONFIG_PREFIX`/`npm_config_prefix` env variables must be Windows-like paths for the `npm install -g` to pick the right path with Cygwin.

There is a PR to fix it:
https://github.com/pre-commit/pre-commit/pull/612
